Client/WebAuthN/WebAuthnRegisterRequest.swift

```diff
@@ -0,0 +1,90 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+import Shared
+
+struct WebAuthnRegisterRequest {
```

Comment:
Overall I'd say this is a bit hard to follow, and I'd recommend smaller structs, with compiler synthesized `Codable`, based on the JSON structure rather than a single flat struct, unless you have a specific reason for it?

You only seem to do anything custom with the `pubKeyAlg` and possibly the `AuthenticatorSelection` so it should be simple.

Example:

```swift
struct WebAuthnRegisterRequest: Decodable {
  struct PublicKey: Decodable {
    struct AuthenticatorSection: Decodable {
      let requireResidentKey: Bool
      let userVerification: String
    }
    struct PublicKeyCredentialParameters: Decodable {
      var alg: String
      var type: String
    }
    struct User: Decodable {
      var displayName: String
      var name: String
      var id: String
    }
    let attestation: String
    let authenticatorSection: AuthenticatorSection?
    let pubKeyCredParams: [PublicKeyCredentialParameters]
    ...
    ...
    let user: User
  }
  let publicKey: PublicKey
}
```